x-pack/plugin/security/qa/basic-enable-security/build.gradle

```diff
@@ -1,76 +0,0 @@
-import org.elasticsearch.gradle.testclusters.StandaloneRestIntegTestTask
-import org.elasticsearch.gradle.internal.test.rest.JavaRestTestPlugin
-import org.elasticsearch.gradle.internal.info.BuildParams
-
-apply plugin: 'elasticsearch.java-rest-test'
-
-//randomise between implicitly and explicitly disabled security
-boolean implicitlyDisabledSecurity = (new Random(Long.parseUnsignedLong(BuildParams.testSeed.tokenize(':').get(0), 16))).nextBoolean()
-
-dependencies {
-  javaRestTestImplementation(testArtifact(project(xpackModule('security'))))
-  javaRestTestImplementation(testArtifact(project(xpackModule('core'))))
-}
-
-tasks.named("javaRestTest").configure {
-  description = "Run tests against a cluster that doesn't have security"
-  if (!implicitlyDisabledSecurity) {
-    systemProperty 'tests.has_security', 'false'
-  }
-}
-
-if (BuildParams.inFipsJvm){
-  // This test cluster is using a BASIC license and FIPS 140 mode is not supported in BASIC
-  tasks.named("javaRestTest").configure{enabled = false }
-}
-
-testClusters {
-  javaRestTest {
-    testDistribution = 'DEFAULT'
-    numberOfNodes = 2
-    setting 'xpack.ml.enabled', 'false'
-    setting 'xpack.license.self_generated.type', 'basic'
-    if (!implicitlyDisabledSecurity) {
-      setting 'xpack.security.enabled', 'false'
-    }
-  }
-}
-
-tasks.register("javaRestTestWithSecurityEnabled", StandaloneRestIntegTestTask) {
-  mustRunAfter("javaRestTest")
```

Comment:
I would prefer we don't remove this QA test.

I looked at the other QA tests (they need some care, but this is a different conversation, I'll take care of it) and this is the only one that deals with enabling security after a restart and we still support running with security disabled.

Maybe now it would be more valuable a QA test where the cluster switches from security enabled to disabled ( which is a less potentially troublesome action, so I'm not pleading for it), but it's still valuable, I think, that a node can start with Security enabled the first time after a restart with a non-empty data dir.

Reply:
> QA tests (they need some care, but this is a different conversation, I'll take care of it)

I now see you've also removed `x-pack/plugin/security/qa/security-not-enabled` which was redundant because of `x-pack/plugin/security/qa/security-disabled` 👍

Reply:
I also remember there's some special behavior palnned for restarts from security disabled from 7.x on basic to security automatically enabled to 8. This QA test could also cover that the said behavior doesn't kick in for every restart to security enabled (only for the upgrade scenario).

Reply:
>  I think, that a node can start with Security enabled the first time after a restart with a non-empty data dir.

Good call, I agree that this has value. I'll add this back and adjust the test

Reply:
Re-enabled it in 00758afd5d6e5b048452005b8795e21a5b6a4574